### DISCUSSION TITLE
Roborock E4 support

### DISCUSSION THREAD

**Discussion by AndreiGr** [PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor#267] [Q&A]
Hi there guys,

Is Roborock E4 supported under any of the available map formats?
If yes, can you let me know which one I should use?

**Comment by PiotrMachowski** [comment]
It should be discovered automatically, if not you can try each one manually

**Reply by AndreiGr** [reply]
Well, currently I only see this:
![image](https://user-images.githubusercontent.com/10116768/160253315-32fcfa28-6d02-4a5c-93bf-f56eef3aa807.png)

How/where can I modify which map format it is using?
Or, is it something else that might be wrong?

**Reply by PiotrMachowski** [reply]
Do you know model code of your device? (something like `roborock.vacuum.s5`)

**Reply by AndreiGr** [reply]
roborock.vacuum.a01

**Reply by losza** [reply]
Hello. I have the same problem. Will it be add support for Roborock E4 (`roborock.vacuum.a01`)?
